tests/test_distances.py

```diff
@@ -92,9 +92,9 @@ def test_invalid_geocoding_dict():
     }
 
     # Check if raises error
-    with pytest.raises(AssertionError) as e:
-        is_valid_geocoding_dict(loc_dict)
-    assert e.type is AssertionError
+    with pytest.raises(ValidationError) as e:
```

Comment:
What exactly is the difference between an assertion error and a validation error and what area reasons to raise a validation error instead of an assertion error?

Reply:
An assertion error is raised when an `assert` statement fails
The `ValidationError` is a pydantic error (see the beginning of the script, where it is imported) which is raised when the validation of an object with a pydantic model was not successful. 

In this test, I changed the logic - we are not calling the custom function `is_valid_geocoding_dict()` anymore, because this validation is now done with pydantic using the Model specification `StructuredLocation`